pywaspgen/burst_datagen.py

```diff
@@ -94,7 +108,7 @@ def gen_burstlist(self, burst_idx=-1):
                 burst_list.append(candidate_burst)
                 sig_count += 1
             elif not self.config["spectrum"]["allow_collisions_flag"]:
-                if not self.__check_collisions(candidate_burst, burst_list):
+                if not self.__check_collisions(candidate_burst, burst_list) and not self.__check_not_observed(candidate_burst):
```

Comment:
This logic will only prevent the burst from being added when the collisions flag is off. This check needs to go around all cases.